stl/inc/memory

```diff
@@ -309,15 +309,15 @@ _NoThrowFwdIt uninitialized_fill_n(_NoThrowFwdIt _First, const _Diff _Count_raw,
 #endif // _HAS_IF_CONSTEXPR
 
 #if _HAS_CXX20
-template <class _Ty, class... _Args_pack>
-/* constexpr */ _Ty* construct_at(_Ty* const _Location, _Args_pack&&... _Args) {
-    return ::new (static_cast<void*>(_Location)) _Ty(_STD forward<_Args_pack>(_Args)...);
+template <class _Ty, class... _Types>
+/* constexpr */ _Ty* construct_at(_Ty* const _Location, _Types&&... _Args) {
+    return ::new (const_cast<void*>(static_cast<const volatile void*>(_Location))) _Ty(_STD forward<_Types>(_Args)...);
 }
 
 namespace ranges {
-    template <class _Ty, class... _Args_pack>
-    /* constexpr */ _Ty* construct_at(_Ty* const _Location, _Args_pack&&... _Args) {
-        return ::new (static_cast<void*>(_Location)) _Ty(_STD forward<_Args_pack>(_Args)...);
+    template <class _Ty, class... _Types>
+    /* constexpr */ _Ty* construct_at(_Ty* const _Location, _Types&&... _Args) {
+        return _STD construct_at<_Ty, _Types...>(_Location, _STD forward<_Types>(_Args)...);

```

Comment:
Does nobody think about DEBUG performance? 

No seriously, as far as I know if you can avoid an additional function call do it. That is one of those places where coding in the STL is diametrically different from how I would do it at work but I can fully understand the reasoning.

Reply:
Performance is not something you really need in debug mode, is it? Besides if this function call wouldn't be inlined it would probably be a basic `jmp` in assembly as these functions have the same arguments (although I'm not an expert in that field so please correct me if I'm wrong). But I can understand you because I like to enable at least basic inlining of `inline` functions in debug mode.

Reply:
We (and our customers) think performance is something you need in debug mode.

Reply:
Ok